rust_src/src/chartable.rs

```diff
@@ -235,11 +235,7 @@ pub fn set_char_table_parent(mut chartable: LispCharTableRef, parent: Option<Lis
         }
     }
 
-    chartable.parent = if let Some(p) = parent {
-        p.as_lisp_obj()
-    } else {
-        Qnil
-    };
+    chartable.parent = if let Some(p) = parent { p.into() } else { Qnil };
```

Comment:
```suggestion
    chartable.parent = parent.map_or(Qnil, |p| p.into());
```

I think the `map_or` makes more sense. But if not, wouldn't `match` be a better fit? Not on you, this was existing code. I think we over used the `if let Some()` idiom.

Reply:
We can do even better than that: `parent: Option<LispCharTableRef>`, so all we need is `chartable.parent = parent.into()`.